packages/build-tools/bin/rivet.sh

```
@@ -65,20 +65,50 @@ get_foundry_version() {
-    source ~/.bashrc 2>/dev/null || true
-    source ~/.zshrc 2>/dev/null || true
-    
-    # Try to run foundryup
-    if command -v foundryup >/dev/null 2>&1; then
-        foundryup
+
+    # Detect shell and source appropriate rc file
+    # Since this script runs under sh, we need to detect the user's actual shell
+    if is_user_shell_zsh; then
+        echo "🔧 ZSH detected. Sourcing ~/.zshenv..."
+        source ~/.zshenv 2>/dev/null || true
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)